### PR TITLE
boards: arm: change display driver to ltdc on stm32f429i_disc1

### LINKED ISSUE #68103: Display: STM32: display in not working on stm32f429i_disc1
**Describe the bug**
The drivers/display sample gives random noise on the screen. LVGL samples, modules/lvgl/demos & subsys/display/lvgl, give distorted frames.

 - What target platform are you using? stm32f429i_disc1 board
 - What have you tried to diagnose or workaround this issue? Found a fix to ili9341 driver that must be applied on top of PR #66271.
   PR #66271 alone gives a white screen.
 - Is this a regression? No info

**To Reproduce**
1. cd ~/zephyrproject/zephyr
2. west build -p always -b stm32f429i_disc1 samples/drivers/display
3. west flash

**Expected behavior**
Correct display of frames.

**Impact**
showstopper

**Logs and console output**
N/A

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: zephyr-sdk-0.16.3
 - Commit SHA or Version used: main, 571b647825b0402b1b46b4347b1db508b14cdf88

**Additional context**
This issue needs to be solved to be able to test PR #64587 on stm32f429i_disc1.

### PR DESCRIPTION
**Pull request by ZhuZhongjie**
Change display driver to ltdc on stm32f429i_disc1 board.

### CODE REVIEW

**Comment by erwango** [comment]
@ajarmouni-st Can you test it is OK ?

**Comment by ajarmouni-st** [comment]
> Change display driver to ltdc on stm32f429i_disc1 board.

Can you elaborate why this change is needed?

**Comment by ZhuZhongjie** [comment]
> > Change display driver to ltdc on stm32f429i_disc1 board.
> 
> @ZhuZhongjie Can you elaborate why this change is needed?
The LCD on the  stm32F429IDISCOVERY Discovery kit is using the RGB protocol, so this should change to the lcdc driver, the spi interface connected to the ILI9341 LCD controller is used to initial the LCD.

**Comment by ZhuZhongjie** [comment]
> @ajarmouni-st Can you test it is OK ?

Yes, it is OK in my stm32f429i Discovery kit board

**Comment by ajarmouni-st** [comment]
> Yes, it is OK in my stm32f429i Discovery kit board
I tested with sample/drivers/display & I get nothing with your modif, but with &ili9341 I have some output on the display, although I don't think it's exactly what the sample is meant to output.

**Comment by ajarmouni-st** [comment]
> > The LCD on the  stm32F429IDISCOVERY Discovery kit is using the RGB protocol, so this should change to the ltdc driver, the spi interface connected to the ILI9341 LCD controller is used to initialize the LCD.

@ZhuZhongjie In page 7 of https://cdn-shop.adafruit.com/datasheets/ILI9341.pdf , you can see that the ILI9341 controller supports 3 interfaces inc. serial interface.

My understanding is that in order to use the RGB protocol with ILI9341, the ILI9341 driver needs to be modified to support it.

File: boards/arm/stm32f429i_disc1/stm32f429i_disc1.dts [hunk 1]
```diff
@@ -20,7 +20,7 @@
 		zephyr,sram = &sram0;
 		zephyr,flash = &flash0;
 		zephyr,ccm = &ccm0;
-		zephyr,display = &ili9341;
+		zephyr,display = &ltdc;
```
Comment (erwango) [inline boards/arm/stm32f429i_disc1/stm32f429i_disc1.dts:23]: This modification is not sufficient.
Node should also be enabled, maybe pins configured, ...

**Review by erwango — CHANGES_REQUESTED** [review]

**Comment by ZhuZhongjie** [comment]
> > > The LCD on the  stm32F429IDISCOVERY Discovery kit is using the RGB protocol, so this should change to the ltdc driver, the spi interface connected to the ILI9341 LCD controller is used to initialize the LCD.
> 
> @ZhuZhongjie In page 7 of https://cdn-shop.adafruit.com/datasheets/ILI9341.pdf , you can see that the ILI9341 controller supports 3 interfaces inc. serial interface.
> 
> My understanding is that in order to use the RGB protocol with ILI9341, the ILI9341 driver needs to be modified to support it.

https://www.st.com/content/ccc/resource/technical/document/user_manual/6b/25/05/23/a9/45/4d/6a/DM00093903.pdf/files/DM00093903.pdf/jcr:content/translations/en.DM00093903.pdf
image
the section 6.9 the lcd is using RGB protocol.

**Comment by ajarmouni-st** [comment]
> Yes, it is OK in my stm32f429i Discovery kit board

Did you test using samples/drivers/display or something else?

**Review by ajarmouni-st — APPROVED** [review]

**Comment by ajarmouni-st** [comment]
> > Yes, it is OK in my stm32f429i Discovery kit board
> 
> @ZhuZhongjie Did you test using samples/drivers/display or something else?

@ZhuZhongjie I was able to validate that this change is needed using the samples /driver/display /modules/lvgl/demos & /subsys/display/lvgl, but only after patching display_ili9341.c (will open a PR). Without my patch, we get a white screen.

**Review by erwango — APPROVED** [review]

**Pull request merged**